legend-engine-xt-javaPlatformBinding-pure/src/main/resources/core_java_platform_binding/legendJavaPlatformBinding/executionPlanNodes/graphFetch/inMemory/graphFetchInMemory.pure

```diff
@@ -724,11 +724,18 @@ function <<access.private>> meta::pure::mapping::modelToModel::executionPlan::pl
    let fValuesField = javaField(['private', 'static', 'final'], $mapType, 'values', 'initValues()');
    let fValues      = $class->j_field($fValuesField);
    let values       = j_variable($mapType, 'values');
-   
+
    let addValues = $em.enumValueMappings->map({evm |
       $evm.sourceValues
          ->map(sv | ^InstanceValue(genericType = ^GenericType(rawType = $sv->type()), multiplicity = PureOne, values = $sv)->evaluateAndDeactivate()->generateJava($conventions, $debug->indent()))
-         ->map(sv | $values->j_invoke('put', [$sv, $enumType->j_field($evm.enum.name)]))
+         ->map(sv | if($sv->instanceOf(meta::external::language::java::metamodel::Code) && $sv.type->instanceOf(meta::external::language::java::metamodel::Enumeration),
+                    | let sourceEnum = $sv->cast(@StaticFieldAccess).class;
+                      let sourceEnumField = j_string($sv->cast(@StaticFieldAccess).fieldName);
+                      $values->j_invoke('put', [$sourceEnum->j_invoke('getEnumFromPureName', $sourceEnumField, $sourceEnum), $enumType->j_invoke('getEnumFromPureName', j_string($evm.enum.name), $enumType)]);,
+                    | $values->j_invoke('put', [$sv, $enumType->j_invoke('getEnumFromPureName', j_string($evm.enum.name), $enumType)])
+                    );
+              )
+
```

Comment:
In enum mapping source value can be an list of enums, strings etc.
So in case of enums being a source value I am using getEnumFromPureName to get the enum corresponding to that Pure enum. And if the source value is not an enum, I am passing it as it is.

Reply:
As discussed offline rather than handling for enum seperately, I have changed the java generation of enum in the file expressionGeneration.pure.